modules/eks/argocd/main.tf

```diff
@@ -0,0 +1,241 @@
+  host                              = var.host != "" ? var.host : format("%s.%s", var.alb_name, local.regional_service_discovery_domain)
+  enable_argo_workflows_auth        = local.saml_enabled && var.argo_enable_workflows_auth
+  enable_argo_workflows_auth_count  = local.enable_argo_workflows_auth ? 1 : 0
+  argo_workflows_host               = "${var.argo_workflows_name}.${local.regional_service_discovery_domain}"
+
+  oidc_config_map = local.oidc_enabled ? {
+    server : {
+      config : {
+        "oidc.config" = <<-EOT
+          name: ${var.oidc_name}
+          issuer: ${var.oidc_issuer}
+          clientID: ${local.oidc_client_id}
+          clientSecret: ${local.oidc_client_secret}
+          requestedScopes: ${var.oidc_requested_scopes}
+          EOT
+      }
+    }
+  } : {}
+
+  saml_config_map = local.saml_enabled ? {
+    server : {
+      config : {
+        "dex.config" = join("\n", [
+          local.dex_config_connectors,
+          local.dex_config_static_clients,
+        ])
+      }
+    }
+  } : {}
+
+  dex_config_connectors = <<-EOT
+    connectors:
+      - type: saml
+        id: okta
+        name: Okta
+        config:
+          ssoURL: ${local.saml_sso_url}
+          caData: ${local.saml_ca}
+          redirectURI: https://${local.host}/api/dex/callback
+          usernameAttr: email
+          emailAttr: email
+          groupsAttr: group
+    EOT
+
+  dex_config_static_clients = local.enable_argo_workflows_auth ? (<<-EOT
+    staticClients:
+      - id: ${data.sops_file.argo_workflows_sops[0].data["spec.secretTemplates.0.stringData.client-id"]}
+        name: Argo Workflow
+        secretEnv: ARGO_WORKFLOWS_SSO_CLIENT_SECRET
+        redirectURIs:
+          - https://${local.argo_workflows_host}/oauth2/callback
+    EOT
+  ) : ""
+
+  post_render_script = local.enable_argo_workflows_auth ? "./resources/kustomize/post-render.sh" : null
+  kustomize_files_values = local.enable_argo_workflows_auth ? {
+    __ignore = {
+      kustomize_files = { for f in fileset("./resources/kustomize", "[^_]*.{sh,yaml}") : f => filesha256("./resources/kustomize/${f}") }
+    }
+  } : {}
+}
+
+module "argo-workflows-sops" {
+  source  = "../sops-secrets-operator/modules/sops_manifest"
+  enabled = local.enable_argo_workflows_auth
+
+  context              = module.introspection.context
+  kubernetes_namespace = var.kubernetes_namespace
+  sops_directory       = abspath("./resources")
+  sops_file_prefix     = "argo-workflows-keys"
+}
+
+module "argocd-notifications-sops" {
+  source = "../sops-secrets-operator/modules/sops_manifest"
+
+  context              = module.introspection.context
+  kubernetes_namespace = var.kubernetes_namespace
+  sops_directory       = abspath("./resources")
+  sops_file_name       = "argocd-notifications-secret.sops.yaml"
+}
+
+module "argocd" {
+  source  = "cloudposse/helm-release/aws"
+  version = "0.3.0"
+
+  name                   = "" # avoids hitting length restrictions on IAM Role names
```

Comment:
@nitrocode what does your argocd chart install end up getting named? Due to the code [here](https://github.com/cloudposse/terraform-aws-helm-release/blob/master/main.tf#L41), we get funky names for our argocd installation like `argo-cd-argocd-server`, which I am not a fan of. What do your K8s resource names look like with this chart and have you found a way around that name duplication?

Reply:
Ah that's a good question. Im unsure tbh

Reply:
@nitrocode it's that you folks override `fullnameOverride` for all of your chart deployments I believe. This does away with whatever is passed in as the name 👍